### PR TITLE
Adding metrics and alerts for network transmit errors

### PR DESCRIPTION
**Pull request by computate**

### CODE REVIEW

File: cluster-scope/base/core/configmaps/thanos-ruler-custom-rules/configmap.yaml [hunk 1]
```diff
@@ -72,6 +72,18 @@ data:
             PersistentVolumeClaim: "{{ $labels.persistentvolumeclaim }}"
             severity: warning
 
+        - alert: CustomNetworkInterfaceErrors
+          annotations:
+            summary: Node network transmit errors
+            description: "{{ $value }} node network transmit errors occured"
+          expr: node_network_transmit_errs_total > 0
```
Comment (aabaris) [inline cluster-scope/base/core/configmaps/thanos-ruler-custom-rules/configmap.yaml]: Sometimes an event occurs causing these counters to be non zero, but without an ongoing problem besides the event.  From there, network interface error counters persist until cleared via reboot.  

I would recommend to change this rule to something like:
expr: increase(node_network_transmit_errs_total[1h]) > 0

**Review by jtriley — APPROVED** [review]

**Review by naved001 — APPROVED** [review]

**Review by aabaris — CHANGES_REQUESTED** [review]
Looks good.  I am making one request to adjust the alert rule (details in my comment for line 79 in configmap.yml) .  Please let me know if you have any questions or would like to discuss.

Thank you.

- Augustine

File: cluster-scope/base/core/configmaps/thanos-ruler-custom-rules/configmap.yaml (hunk 1 above)
Comment (computate) [reply cluster-scope/base/core/configmaps/thanos-ruler-custom-rules/configmap.yaml]: Thanks for the insight @aabaris , I updated my commit with your suggestion!

**Comment by aabaris** [comment]
Looks good.  Thank you!

**Review by aabaris — APPROVED** [review]

**Pull request merged**